### ISSUE TITLE
AI aggro

### ISSUE THREAD

**Issue by Sairii** [aura-project/aura#202]
While doing the gremlin AI I noticed the following:

		HatesNearby(8000);
		HatesBattleStance(1000);

Sometimes without battlestance they instant aggro me.
Also the times appear to be wrong. 8000 Should be a long time for it to aggro me while its in Alert.
Right now the mob is only very short in alert and then switches to aggro very fast.

I noticed this because the gremlins Alert is basically running away with circles for a very long time... really annoying and on Aura they pretty much do a little run and then smack you.

So probably HatesNearby is checking from the moment the monster notices you but not from the time its in alert.
So basically:

Monster takes 4 seconds to go into alert, takes 8+- seconds to build hate (probably official?) =12 seconds
Monster takes 4 seconds to go into alert, takes only 4+- seconds to build hate because it builds it from the start of notice instead of alert start (aura) = 8 Seconds

**Comment by exectails** [comment]
In the following video Gremlins do aggro instantly, once at the beginning, and again at ~1:55:
https://www.youtube.com/watch?v=HMaNT0hYXvc

Right now the hate delay starts with aware, same as the alert delay. I don't remember exactly why I did it like this, but I assume it matched my observations with wolves.

**Comment by Sairii** [comment]
It actually has to do with visual and audio aggro range.
Visual goes into alert, aggro goes straight into hate.
http://aura-project.org/forum/topic/107-monsters-can-hear-you-repost/

**Comment by exectails** [comment]
>Visual goes into alert, aggro goes straight into hate.
>so a bear has 400 around him 360° instant aggro
700 - 400 90° alert

Are you sure you're not mixing things up here...? I've just tested this myself, and I'm only able to get bears into alert if I stay *behind* them, in *audio* range. If they *see* me, they insta aggro. This kinda makes more logical sense as well, as I've said before, in the topic you've linked.

**Comment by exectails** [comment]
After killing some more bears, here's how I think it works:

The nearby hate counter normally starts when they go into alert, which is 6s after they noticed you (when you got into audio range), under Aura's current default rules. This makes sense, because that's the point where they officially "see" you, turning around to you ("oh, there is someone! \*alert\*"). They can't hate you before they notice you.

However, when they actually *see* you, the rules change, they might not go into alert, but they must have noticed you, unless they're blind. I think in this case the hate counter can start before alert, at the point they saw you. This makes sense as well, because again, they start you hate you as soon as they're actually aware of you.

The actual aggroing still seems to take until *after* alert though. Even if visually they only go into alert for a fraction of a second, they do seem to go into alert first.

This might explain all the "randomization" we've experienced, just that it wasn't random at all. If you went behind a bear, and he then turned around, the alert counter was already going, but the hate counter also started, resulting in a potential delay between alert and hate. But only if the bear didn't turn around for at least 2-3s, because that's the difference between the alert and the hate delay.

This also matches my findings with wolves and imps I think. For example, sometimes imps aggroed instantly (after reaching alert), and sometimes it took them 2 seconds. But they only have a 2s hate counter, so naturally they would insta-aggro if they saw me. But they would wait for up to 2s if they didn't "see" me before the alert.

**Comment by Sairii** [comment]
Actually isnt it the other way around?

Visual aggro = they alert
Audio aggro = instant hate

Is what I found.

**Comment by exectails** [comment]
It's not what I experienced yesterday o.o When you manage to stay completely out of sight, behind a bear for example, in audio range, they aggro you? Because that would be the exact opposite of my experience.

**Comment by Sairii** [comment]
>It's not what I experienced yesterday o.o When you manage to stay completely out of sight, behind a bear for example, in audio range, they aggro you? Because that would be the exact opposite of my experience.

OK... well i guess its time for getting smashed by bears V3

**Comment by exectails** [comment]
After some more testing, we've discovered that monsters that doubt players, and hate nearby, don't seem to aggro until you're in audio range.

**Comment by Sairii** [comment]
So yeah its basically 
![wat](https://cloud.githubusercontent.com/assets/10909962/12275532/c72ba010-b970-11e5-9e08-1e47b625f56b.jpg)

Red = Audio Range = Instant Aggro
Purpleish = Visual Range = Alert Zone -> Most alert AIs move closer into the audio range and then aggro.

Basically you can never aggro a direwolf if you alerted it and IMMEDIATLY run away. FAST
I had one following me for quite a while ^^
So yeah its basically 
![wat](https://cloud.githubusercontent.com/assets/10909962/12275532/c72ba010-b970-11e5-9e08-1e47b625f56b.jpg)

Red = Audio Range = Instant Aggro
Purpleish = Visual Range = Alert Zone -> Most alert AIs move closer into the audio range and then aggro.

Basically you can never aggro a direwolf if you alerted it and IMMEDIATLY run away. FAST
I had one following me for quite a while ^^

**Comment by exectails** [comment]
Closing, as I believe this was implemented a while ago. Mentioning #171 for later reference.
